### PR TITLE
Wrap SymEngine classes in Python

### PR DESCRIPTION
**Pull request by isuruf**
I've added Log and Trig Functions.
Let me know if it's okay, so I can start on Hyperbolic Functions.

### CODE REVIEW

File: symengine/python/symengine/lib/symengine_wrapper.pyx [hunk 1]
```diff
@@ -51,6 +47,32 @@ cdef c2py(RCP[const symengine.Basic] o):
...
+    elif (symengine.is_a_ACos(deref(o))):
+        r = ACos.__new__(ACos)
+    elif (symengine.is_a_ATan(deref(o))):
+        r = ATan.__new__(ATan)
+    elif (symengine.is_a_ACot(deref(o))):
+        r = ACot.__new__(ACot)
+    elif (symengine.is_a_ACsc(deref(o))):
+        r = ACsc.__new__(ACsc)
+    elif (symengine.is_a_ASec(deref(o))):
+        r = ASec.__new__(ASec)
```
Comment (certik) [inline symengine/python/symengine/lib/symengine_wrapper.pyx]: This is fine for this PR, but after it's merged, we should just read the type id and then just compare an integer, as opposed to calling all these functions.

**Comment by certik** [comment]
I think that this looks very good. I went over it and have no fundamental issues with this. There is couple of ways how these things can be wrapped in Python (i.e. in sympy we just have classes like `cos`, and `cos(0)` then returns 0 using the `__new__` method, here instead we have a `cos` function that returns 0 or a `Cos` instance --- which I think makes more sense for SymEngine), and I think the way it is done in this PR should work well for now.

**Comment by isuruf** [comment]
Added wrappers for hyperbolic functions and simplified the code a little bit

**Comment by certik** [comment]
Looks good.

**Comment by certik** [comment]
+1 from me to merge.

**Pull request merged**